polaris/dataset/_competition_dataset.py

```diff
@@ -13,7 +14,7 @@ class CompetitionDataset(DatasetV1):
     """
 
     @model_validator(mode="after")
-    def _validate_model(self):
+    def _validate_model(self) -> Self:
```

Comment:
Is `self` implicitly returned from `@model_validator`? We don't do it explicitly here so the `Self` typing may not be applicable.

Reply:
A better way to put it, could we add `return self` at the end of this validator?

Reply:
The methods needs to `return self`at the end.

Reply:
Good catch! Thanks! 🙏